backend/src/test/java/com/hackner/musicompass/service/ArtistServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package com.hackner.musicompass.service;
+
+import com.hackner.musicompass.db.ArtistMongoDb;
+import com.hackner.musicompass.discogsapi.model.DiscogsArtist;
+import com.hackner.musicompass.discogsapi.model.DiscogsArtistSearchResults;
+import com.hackner.musicompass.discogsapi.service.DiscogsArtistApiService;
+import com.hackner.musicompass.model.Artist;
+import com.hackner.musicompass.model.ArtistInfo;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.sql.Date;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.*;
+
+class ArtistServiceTest {
+
+    private final DiscogsArtistApiService discogsArtistApiService = mock(DiscogsArtistApiService.class);
+    private final ArtistMongoDb artistMongoDb = mock(ArtistMongoDb.class);
+    private final TimeUtils timeUtils = mock(TimeUtils.class);
+    private final ArtistReleaseService artistReleaseService = mock(ArtistReleaseService.class);
+    private final ArtistService artistService = new ArtistService(discogsArtistApiService, artistReleaseService, artistMongoDb, timeUtils);
+
+    @Test
+    @DisplayName("Get Artist who is already saved in DB")
+    public void getArtistByNameFromDb (){
+        //GIVEN
+        String artistName = "Hans Hammer";
+
+        Artist artist = Artist.builder()
+                .artistName(artistName)
+                .artistInfo(new ArtistInfo().builder()
+                        .artistImageUrl("HammerImageUrl")
+                        .discogsArtistId("111")
+                        .discogsArtistUrl("HammerUrl")
+                        .build())
+                .build();
+
+        when(artistMongoDb.existsById(artistName)).thenReturn(true);
+        when(artistMongoDb.findById(artistName)).thenReturn(Optional.of(artist));
+
+        //WHEN
+        Artist actual = artistService.getArtistByName(artistName);
+
+        //THEN
+        assertThat(actual, equalTo(artist));
```

Comment:
maybe add a verify() for your mocked methods to ensure they are invoked in the method context

Reply:
done